src/main.py

```diff
@@ -0,0 +1,54 @@
+import argparse
+
+from pipeline.eval import eval_on_records
+from utils import get_steps_right
+
+def main(args):
+    if args.load_in_8bit is True \
+        and args.load_in_4bit is True:
+        raise ValueError("both load_in_8bit and load_in_4bit are set. "
+                         "only one of them should be set at a time")
+    
+    valid, input_steps = get_steps_right(args.steps)
```

Comment:
What does `get_steps_right()` mean?

Reply:
the arg `--steps` can take steps such as ft(fine-tuning), eval(evaluation), synth-gen(synthetic data generation), deploy(deployment). However, we can't assume the input is given in a valid form in terms of order, duplications, ...

`get_steps_right()` makes sure if everything is right in that manner.

Reply:
`validate_steps()` might be a better name?